Add pie chart to visualize slur distribution by severity using D3.js

Mahimahto opened the pull request:
### PR Description
This pull request implements a pie chart visualization on the Dashboard page to show the distribution of crowdsourced slurs based on their `level_of_severity` field (low, medium, high). The chart is built using D3.js and is displayed alongside the existing bar chart.

**Features Added :-**

- Created a new D3-based pie chart component to represent severity-level distribution.
- Integrated the pie chart into the existing dashboard layout beside the bar chart.

**Ecto Functions Used :-**
To prepare the severity data for the pie chart, the following Ecto functions were used:
```
def get_severity_distribution do
  UliCommunity.UserContribution.CrowdsourcedSlur
  |> group_by([s], s.level_of_severity)
  |> select([s], %{label: s.level_of_severity, count: count(s.id)})
  |> Repo.all()
end
```
**Explanation :-**

- `group_by/2` groups the slurs by their `level_of_severity` field.
- `select/2` returns a map with the label (severity level) and count (number of slurs in that group).
- `Repo.all/1` executes the query and returns a list of maps, each representing a severity level and its count.
- This data is passed to the frontend and rendered using `D3.js` to produce a clean, informative pie chart.

**Screenshot :-**
![Screenshot from 2025-06-16 15-48-23](https://github.com/user-attachments/assets/65dca3da-7649-4096-837d-ad78f766885c)

**Related Issue :-**
This PR addresses [Issue #789](https://github.com/tattle-made/Uli/issues/789).

aatmanvaidya commented on uli-community/lib/uli_community_web/live/dashboard_live.ex:
```diff
@@ -6,12 +6,17 @@ defmodule UliCommunityWeb.DashboardLive do
   def mount(_params, _session, socket) do
     if connected?(socket), do: Phoenix.PubSub.subscribe(UliCommunity.PubSub, @topic)
+
     slurs = UserContribution.get_top_slurs(10)
-    {:ok, assign(socket, slurs: slurs)}
+    severity_data = UserContribution.get_severity_distribution()
```
@dennyabrain wanted to ask a small question here

right now in the mount() function, we are calling 2 function which are making some sort of REST API call to the DB. 
```elixir
    slurs = UserContribution.get_top_slurs(10)
    severity_data = UserContribution.get_severity_distribution()
```

what is the best/recommend way to do this in a liveview? is ti fine to have 2 functions making ecto calls in mount() and how to add some error handling here?

Mahimahto commented:
I’ve added error handling in the `DashboardLive` file:

- Wrapped both `get_top_slurs/1` and `get_severity_distribution/0` in `try/rescue` blocks (in `mount` and `handle_info`)
- If there is any error, it assigns `nil` instead of crashing the LiveView
- Added `Logger.error` to log the error messages
- Updated the `.heex `file to check if data is not `nil` before showing the chart
- If data is `nil`, it shows the message "Unable to load data"

dennyabrain commented on uli-community/lib/uli_community_web/live/dashboard_live.ex:
```diff
@@ -7,15 +7,37 @@ defmodule UliCommunityWeb.DashboardLive do
   def handle_info(:slur_changed, socket) do
-    slurs = UserContribution.get_top_slurs(10)
-    severity_data = UserContribution.get_severity_distribution()
+    slurs =
```
@aatmanvaidya 
We don't have to build realtime components for every feature. This page does not need to react to any changes.
